### ISSUE TITLE
Phone crashes when opening "What's New" in the latest two versions

### ISSUE THREAD

**Issue by d-albrecht** [TeamNewPipe/NewPipe#3255]
First of,: Awesome app for any phone especially for phones without those Google-Bloatware-Apps.

Today I updated from 0.18.4 to 0.18.7 and immediately ran into issues opening the "What's New" tab. Every time I try to do so, the loading circle starts (as usual), after a second or two the animation stops moving but stays visible. again a few seconds later my phone (not just the app) crashes completely and automatically reboots.

As you can see, I'm not able to submit any regular crash reports, because the app never shows me one, instead the phone shuts down completely. But if you know things I can test to further narrow down the problem, please tell me.

I use NewPipe for a few months now and this error never occurred previously. The described behavior also happens with the version 0.18.6 (I tried it out, too) that I had skipped (because not updating f-droid that often). I haven't tried 0.18.5 yet as it isn't listed in f-droid. Do you thing, is it worth testing this version?

I'm running a Fairphone model 2 with Android 7.1.2.

**Comment by d-albrecht** [comment]
Okay, out of personal curiosity, I just tried the 0.18.5 without you asking for and it works just fine as far as I can tell. I will stay on 0.18.5 for now and test for any hiccups. But as the mentioned issue occurs consistently EVERY time I tried it (with both tested versions around a dozen times), the 0.18.5 seems to not have this issue. Therefore, version 0.18.6 introduced it apparently.

**Comment by Stypox** [comment]
>  I'm not able to submit any regular crash reports, because the app never shows me one, instead the phone shuts down completely. But if you know things I can test to further narrow down the problem, please tell me.
Really strange. Is there a way you can get some data via adb logcat?

**Comment by d-albrecht** [comment]
I can look into this again the next days, but so far I wasn't able to get adb (running and) connecting to my phone today.

In the meantime, are you aware of any changes between these two releases that could cause such a tremendous change in behavior of the app that Android is caused to crash?

At the moment, I'm writing a similar post for the Fairphone forum to see if others can report similar problems, so perhaps this can help figure out the actual problem. If this is a device-/OS-specific problem and given that Fairphone is a niche brand, the Fairphone community might be really helpful here.

**Comment by wb9688** [comment]
https://github.com/TeamNewPipe/NewPipe/compare/v0.18.5...v0.18.6 are the only changes. Does the APK from #3240 crash? If not, it's fixed in the next release anyway, which features a much improved "What's New" tab.

**Comment by d-albrecht** [comment]
You're awesome! Not only does this work for me, this also brings exactly the "What's New"-tab view I would have asked for shortly after. This is such an improvement over the current one. So, in hope that this apk behaves exactly as the final NewPipe version will, I think this issue is resolved, although none of us knows the actual cause. ;)

**Comment by d-albrecht** [comment]
I will have to observe this further, I fear, as I had another phone crash with the preview apk just minutes ago.

For some context, I had imported my subscriptions from the regular 0.18.5 release and by the time I first imported this subscription list and polled for the "What's New" tab everything worked well. Today, I revisited this tab and realized that the view isn't auto-updating anymore. Essentially, that's fine with me, although I hope you will communicate this in the release notes.

But knowing that the feed wasn't up to date, I reloaded one of my sub-lists (again awesome feature) and it worked fine except for a small period of time where the loading animation had stopped moving again (like in the initial issue description), but it resumed after some seconds and when the loading animation was gone, the list seemed updated. After that I went to the full list and tried an update and the phone crashed/rebooted. When the phone was back up, I tried to reload each of the remaining sub-lists and they all updated but again with some seconds of stopped animation-movement. Thereafter the full list recognized the updates and didn't allow further tries.

My observation with this is, that the period of time where the animation stopped seemed related to the number of feeds in this sub-list. The more elements to poll, the longer stopped the animation (but as I could only try this once right now, this might also be a subjective impression). Naturally, as the full list has the most feeds in it the problem would be worst here and given that the earlier app-versions only offered an update-poll for the full list, this would explain why it happened each time with those versions. I begin to suspect that my issues depends in the number of subscriptions. If the app tries to update too many feeds at the same time, it crashes the phone.

I haven't managed to adb-connect my phone yet, but given that the issue persists, I'll try again tomorrow and reload my feeds as soon as the app permits to do so. But perhaps my observation can help to identify the issue already.

**Comment by wb9688** [comment]
Could you reproduce this when you remove data from that debug APK and add subscriptions manually? If not, would you mind sharing the export from v0.18.5 that caused this? PS: in Settings -> Content there's an option to lower the feed update threshold. Don't refresh your feed to often though, as then YouTube might think you're a bot and wants you to solve some ReCaptchas.

**Comment by d-albrecht** [comment]
To test with some manually added subscriptions I took the top 15 or so trending videos and added their channels. I wsn't able to poll them yet as visiting all the channels already polled them one by one. So in a few hours or so (?) I can tell you whether not importing helps with this. But I suspect, because while visiting the top trending channels some of the channels have shown a similar (although only short) moment of paused animation. Apparently, one channel can be large enough for a stopped animation. This backs up my assumption, I'd say.

**Comment by d-albrecht** [comment]
I deleted two entries that I don't feel especially comfortable with posting but this the otherwise complete original subscription list:
`{"app_version":"0.18.4","app_version_int":840,"subscriptions":[{"service_id":0,"url":"https://www.youtube.com/channel/UCdBK94H6oZT2Q7l0-b0xmMg","name":"ShortCircuit"},{"service_id":0,"url":"https://www.youtube.com/channel/UCjMZjGhrFq_4llVS_x2XJ_w","name":"Fantasy & World Music by the Fiechters"},{"service_id":0,"url":"https://www.youtube.com/channel/UCwF4bAx_W482_AQu-mv1vWg","name":"Traditional Music by Ean Grimm & the Fiechters"},{"service_id":0,"url":"https://www.youtube.com/channel/UCLLnIcQll56hJ89dDin4UcA","name":"Medieval & World Music by Brandon Fiechter"},{"service_id":0,"url":"https://www.youtube.com/channel/UCXuqSBlHAE6Xw-yeJA0Tunw","name":"Linus Tech Tips"},{"service_id":0,"url":"https://www.youtube.com/channel/UCVRJdMxf_nuCR6C1ZMgn7Ug","name":"Fantasy & Dark Music by Derek Fiechter"},{"service_id":0,"url":"https://www.youtube.com/channel/UCt-oJR5teQIjOAxCmIQvcgA","name":"Carpool Critics"},{"service_id":0,"url":"https://www.youtube.com/channel/UCBZiUUYeLfS5rIj4TQvgSvA","name":"Channel Super Fun"},{"service_id":0,"url":"https://www.youtube.com/channel/UC0vBXGSyV14uvJ4hECDOl0Q","name":"Techquickie"},{"service_id":0,"url":"https://www.youtube.com/channel/UCeeFfhMcJa1kjtfZAGskOCA","name":"TechLinked"},{"service_id":0,"url":"https://www.youtube.com/channel/UCaBf1a-dpIsw8OxqH4ki2Kg","name":"Geek & Sundry"},{"service_id":0,"url":"https://www.youtube.com/channel/UCL81YHgzH8tcrFfOJJwlSQw","name":"LinusCatTips"},{"service_id":0,"url":"https://www.youtube.com/channel/UCg6gPGh8HU2U01vaFCAsvmQ","name":"Chris Titus Tech"}]}`
And importing a subscription list can cause the same crashes I just learned.

**Comment by wb9688** [comment]
Import indeed causes the subscriptions to be polled, if they aren't cached. I can't reproduce it when importing the subscription list you provided. I think we really need adb logcat. What problems are you facing with adb? If using Linux, you need to set up some udev rules first.

**Comment by d-albrecht** [comment]
Running Windows, adb is extracted and is executable but running `adb devices` does nothing (meaningful): no option on my phone to accept the PC as debugging device, adb always shows an empty list and this all regardless of the connection mode (MIDI, PTP, MTP, charge-only). My charging cable might just not be right for this, some stackexchange-thread suggests.

**Comment by wb9688** [comment]
You have enabled adb ("USB Debugging" is what that option is called iirc) in Settings -> Developer Options, right?

**Comment by d-albrecht** [comment]
Sure, I have ;) but thanks for thinking about such early points of failure. I even re-downloaded adb to cancel out any issues I could think of, but no success so far.

**Comment by wb9688** [comment]
Have you installed your OEM's driver? See https://developer.android.com/studio/run/oem-usb

**Comment by d-albrecht** [comment]
1. Fairphone doesn't provide specific usb-drivers. They point to generic drivers but I think I installed them when I got the phone. But that doesn't matter because of point 2.

2. As of today I'm not able anymore to connect my phone to ANY (!) PC. While charging with (different) cables at an AC charger is still possible, plugging my phone into any PC with any cable at best gets me a charging indication, at worst not even shows up as charging. I'm totally puzzled right now, because yesterday at least file transfer (and of course charging) was still possible and I haven't changed anything on both my phone and my PC. And as I said I even tried other PCs with the same (or similar) results while other phones still connect just fine. I don't know whether my connection problems and the phone crashes are related, but right now I'm unable to provide additional information until my connection problems are resolved. And I don't know when this will be, as I don't even have a clue where to look for.

**Comment by Stypox** [comment]
@d-albrecht try to use this app to read logcat, though I don't know if it still saves logs even after a reboot: https://www.f-droid.org/en/packages/com.dp.logcatapp/

**Comment by d-albrecht** [comment]
nice app, only problem, that my preferred method of granting log-permission would have been the adb-way as I'm not really happy with granting apps root-access. I did nonetheless but I can support your thought, that logs don't survive a reboot apparently. The only newpipe related exception that I could observe were the following, but I'm not quite convinced that they caused anything crucial.

<details>
<summary>newpipe logs</summary>

<<< log_count = 3 >>>
[03-25 14:30:05.915 6429:6429 W/Glide]
Load failed for /storage/emulated/0/backup/newpipe_subscriptions_202003061248.json with size [82x82]
class com.bumptech.glide.load.engine.GlideException: Failed to load resource
There were 3 causes:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
  Cause (1 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{FileInputStream->Object->Drawable}, LOCAL
    Cause (1 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->c->Drawable}
    Cause (2 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->Bitmap->Drawable}
    Cause (3 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->BitmapDrawable->Drawable}
  Cause (2 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{ParcelFileDescriptor->Object->Drawable}, LOCAL
There were 2 causes:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
    Cause (1 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{ParcelFileDescriptor->Bitmap->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000
    Cause (2 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{ParcelFileDescriptor->BitmapDrawable->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000
  Cause (3 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{AssetFileDescriptor->Object->Drawable}, LOCAL
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
    Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{AssetFileDescriptor->Bitmap->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000

[03-25 14:30:05.954 6429:6429 W/Glide]
Load failed for /storage/emulated/0/backup/newpipe_subscriptions_202003212333.json with size [82x82]
class com.bumptech.glide.load.engine.GlideException: Failed to load resource
There were 3 causes:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
  Cause (1 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{FileInputStream->Object->Drawable}, LOCAL
    Cause (1 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->c->Drawable}
    Cause (2 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->Bitmap->Drawable}
    Cause (3 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->BitmapDrawable->Drawable}
  Cause (2 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{ParcelFileDescriptor->Object->Drawable}, LOCAL
There were 2 causes:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
    Cause (1 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{ParcelFileDescriptor->Bitmap->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000
    Cause (2 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{ParcelFileDescriptor->BitmapDrawable->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000
  Cause (3 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{AssetFileDescriptor->Object->Drawable}, LOCAL
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
    Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{AssetFileDescriptor->Bitmap->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000

[03-25 14:30:05.973 6429:6429 W/Glide]
Load failed for /storage/emulated/0/backup/NewPipeData-20200322_004657.zip with size [82x82]
class com.bumptech.glide.load.engine.GlideException: Failed to load resource
There were 3 causes:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
  Cause (1 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{FileInputStream->Object->Drawable}, LOCAL
    Cause (1 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->c->Drawable}
    Cause (2 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->Bitmap->Drawable}
    Cause (3 of 3): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{FileInputStream->BitmapDrawable->Drawable}
  Cause (2 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{ParcelFileDescriptor->Object->Drawable}, LOCAL
There were 2 causes:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
    Cause (1 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{ParcelFileDescriptor->Bitmap->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000
    Cause (2 of 2): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{ParcelFileDescriptor->BitmapDrawable->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000
  Cause (3 of 3): class com.bumptech.glide.load.engine.GlideException: Failed LoadPath{AssetFileDescriptor->Object->Drawable}, LOCAL
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
    Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Failed DecodePath{AssetFileDescriptor->Bitmap->Drawable}
There was 1 cause:
java.io.IOException(java.lang.RuntimeException: setDataSource failed: status = 0x80000000)
 call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class java.io.IOException: java.lang.RuntimeException: setDataSource failed: status = 0x80000000

At least they mention the subscription file.

</details>

**Comment by wb9688** [comment]
@d-albrecht: I don't think we're even using Glide, so that's strange. And yeah, you shouldn't send full logs uncensored.

**Comment by adruino-io** [comment]
Had the same issue with an FF2, starting with v.18.6 and persistent to 19.2  Downgraded, worked well. After importing my settings from the updated version (19.1) the problem also occured in 18.5. Just updated to 19.3 yesterday to provide a crash-log for this ticket and everything is working fine again. Not a single crash even when refreshing the "Trends" and "Feeds" like a lunatic. Seems like v19.3 fixed it. Also loading time seems to have improved (seemed to get stuck on occasion before that).

@d-albrecht can you confirm this?

**Comment by d-albrecht** [comment]
To be honest, I was using version 0.18.5 exclusively since I discovered that this was the last working version for me. This had the advantages that F-Droid hasn't suggested any further updates since, as the 0.18.5 version is no official F-Droid build, and I could use NewPipe like I had before (not having to fear that the app would crash my phone again if I pressed the wrong buttons). Always knowing, that the next update to YouTube by Google could potentially break everything.
Thanks @adruino-io for letting me know that a new version with - as far as I can tell at this point in time - a fix for my issue was released (whether intentionally with this bug in mind or not). I will monitor this the next days and close this bug report in case I don't have any more occurrences of the described disruptive behavior (I don't want to again, close this case hastily).

**Comment by d-albrecht** [comment]
Apparently I tested the new version for two weeks now instead of the planned one week, but now I think I can tell you for a fact that even with about 50 channel-subscriptions with two third randomly selected from the trending videos from the time I installed the most recent version (meaning that those channels will most likely have a lot of videos uploaded) each and every channel update worked like a charm.

I still don't know what exactly caused this problem in the first place but I really hope you guys can manage to stick with this working situation. ;-)

But I'd say, we're fine right now, so this issue is closed finally(/again)!
